profiling_router.go

```diff
@@ -0,0 +1,147 @@
+package handlers
+
+import (
+	"context"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"strconv"
+
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/go-utils/logger"
+)
+
+const PprofRoutePrefix = "/debug/pprof"
+
+type profiling struct {
+	enable bool
+	auth   pprofAuthentication
+}
+
+type pprofAuthentication struct {
+	username string
+	password string
+}
+
+func NewProfilingRouter(ctx context.Context) (*Router, error) {
+	log := logger.Get(ctx)
+
+	prof := new(profiling)
+
+	err := prof.initialize()
+	if err != nil {
+		return new(Router), errgo.Notef(err, "fail to initialize pprof profiling")
```

Comment:
In case of error we don't return  a usable value

```suggestion
		return nil, errgo.Notef(err, "fail to initialize pprof profiling")
```